code/game/turfs/floor.dm

```diff
@@ -1,525 +1,160 @@
-						if(FF.is_carpet_floor())
-							diagonalconnect |= 1
-
-				//Southeast
-				if(connectdir & SOUTH && connectdir & EAST)
-					if(istype(get_step(src,SOUTHEAST),/turf/open/floor))
-						var/turf/open/floor/FF = get_step(src,SOUTHEAST)
-						if(FF.is_carpet_floor())
-							diagonalconnect |= 2
-
-				//Northwest
-				if(connectdir & NORTH && connectdir & WEST)
-					if(istype(get_step(src,NORTHWEST),/turf/open/floor))
-						var/turf/open/floor/FF = get_step(src,NORTHWEST)
-						if(FF.is_carpet_floor())
-							diagonalconnect |= 4
-
-				//Southwest
-				if(connectdir & SOUTH && connectdir & WEST)
-					if(istype(get_step(src,SOUTHWEST),/turf/open/floor))
-						var/turf/open/floor/FF = get_step(src,SOUTHWEST)
-						if(FF.is_carpet_floor())
-							diagonalconnect |= 8
-
-				icon_state = "carpet[connectdir]-[diagonalconnect]"
-		shoefootstep = FOOTSTEP_CARPET
-		barefootstep = FOOTSTEP_CARPET
-		mediumxenofootstep = FOOTSTEP_CARPET
-
-	else if(is_wood_floor())
-		if(!broken && !burnt)
-			if(!(icon_state in GLOB.wood_icons))
-				icon_state = "wood"
-		shoefootstep = FOOTSTEP_WOOD
-		barefootstep = FOOTSTEP_WOOD
-		mediumxenofootstep = FOOTSTEP_WOOD
-
-/turf/open/floor/return_siding_icon_state()
-	..()
-	if(is_grass_floor())
-		var/dir_sum = 0
-		for(var/direction in GLOB.cardinals)
-			var/turf/T = get_step(src,direction)
-			if(!(T.is_grass_floor()))
-				dir_sum += direction
-		if(dir_sum)
-			return "wood_siding[dir_sum]"
-		else
-			return 0
-
-/turf/open/floor/proc/gets_drilled()
-	return
-
-/turf/open/floor/proc/break_tile_to_plating()
-	if(!is_plating())
-		make_plating()
-	break_tile()
-
-/turf/open/floor/is_plasteel_floor()
-	if(istype(floor_tile,/obj/item/stack/tile/plasteel))
-		return 1
-	else
-		return 0
-
-/turf/open/floor/is_light_floor()
-	if(istype(floor_tile,/obj/item/stack/tile/light))
-		return 1
-	else
-		return 0
-
-/turf/open/floor/is_grass_floor()
-	if(istype(floor_tile,/obj/item/stack/tile/grass))
-		return 1
-	else
-		return 0
-
-/turf/open/floor/is_wood_floor()
-	if(istype(floor_tile,/obj/item/stack/tile/wood))
-		return 1
+	. = ..()
+	if(broken)
+		icon_state = pick(broken_states())
+	else if(burnt)
+		icon_state = pick(burnt_states())
 	else
-		return 0
+		icon_state = initial(icon_state)
 
-/turf/open/floor/is_carpet_floor()
-	if(istype(floor_tile,/obj/item/stack/tile/carpet))
-		return 1
-	else
-		return 0
+///Returns a list of icon_states to show this turf is broken
+/turf/open/floor/proc/broken_states()
+	return list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
 
-/turf/open/floor/is_plating()
-	if(!floor_tile)
-		return 1
-	return 0
+///Returns a list of icon_states to show this turf is burnt
+/turf/open/floor/proc/burnt_states()
+	return list("floorscorched1", "floorscorched2")
 
+///Breaks the turf
 /turf/open/floor/proc/break_tile()
-	if(!breakable_tile || hull_floor) return
-	if(broken) return
+	if(!breakable_tile || hull_floor)
+		return
+	if(broken)
```

Comment:
Could be combined into one if

Reply:
I prefer two lines, since its early returning for a different reason.